### PR TITLE
feat(pt/tf): add bias changing param/interface

### PR DESCRIPTION
**Pull request by iProzd**
Add bias changing param/interface

For pt/tf, add `training/change_bias_after_training` to change out bias once after training.

For pt, add a separate command `change-bias` to change trained model(pt/pth, multi/single) out bias for specific data:

```
dp change-bias model.pt -s data -n 10 -m change
```

UTs for this feature are still in consideration.

## Summary by CodeRabbit

- **New Features**
  - Added a new subcommand `change-bias` to adjust model output bias in the PyTorch backend.
  - Introduced test cases for changing model biases via new test suite.
  
- **Documentation**
  - Added documentation for the new `change-bias` command, including usage and options.
  - Updated `index.rst` to include a new entry for `change-bias` under the `Model` section.

- **Bug Fixes**
  - Adjusted data handling in `make_stat_input` to limit processing to a specified number of batches.

- **Refactor**
  - Restructured training configuration to include the parameter `change_bias_after_training`.
  - Modularized data requirement handling and bias adjustment functions.

### CODE REVIEW

File: deepmd/pt/entrypoints/main.py [hunk 1]
```diff
@@ -377,6 +387,111 @@ def show(FLAGS):
             log.info(f"The fitting_net parameter is {fitting_net}")
 
 
+def change_bias(FLAGS):
+    if FLAGS.INPUT.split(".")[-1] == "pt":
```
Comment (njzjz) [inline deepmd/pt/entrypoints/main.py:391]: This is unsafe is no dot in the filename. Could you use `endswith`?

File: deepmd/pt/entrypoints/main.py [hunk 2]
```diff
@@ -377,6 +387,111 @@ def show(FLAGS):
             log.info(f"The fitting_net parameter is {fitting_net}")
 
 
+def change_bias(FLAGS):
```
Comment (njzjz) [inline deepmd/pt/entrypoints/main.py:390]: I suggest not passing `FLAGS` but expanding the dict.

Comment (njzjz) [reply deepmd/pt/entrypoints/main.py:390]: Other functions in this file have similar issues but can be handled in a separated PR.

**Review by wanghan-iapcm — CHANGES_REQUESTED** [review]
1. please add unit test. 
2. along with stat from data, shall we allow the user explicitly setting the bias number?

File: deepmd/pt/entrypoints/main.py (hunk 2 above)
Comment (iProzd) [reply deepmd/pt/entrypoints/main.py:390]: Tracked in https://github.com/deepmodeling/deepmd-kit/issues/3934.

**Comment by iProzd** [comment]
@coderabbitai do you have any idea how to add a ut for command `dp --pt change-bias` proposed in this PR?

File: deepmd/main.py [hunk 3]
```diff
@@ -51,6 +51,18 @@ def get_ll(log_level: str) -> int:
...
+def float_list(value_str):
+    """Convert string to a list of float values."""
+    values = value_str.split(",")
+    try:
+        return [float(v) for v in values]
+    except ValueError:
+        raise argparse.ArgumentTypeError(
+            f"'{value_str}' cannot be converted to a list of floats, "
+            f"please use ',' to separate each float value."
+        )
```
Comment (njzjz) [inline deepmd/main.py:63]:
```suggestion
    except ValueError as e:
        raise argparse.ArgumentTypeError(
            f"'{value_str}' cannot be converted to a list of floats, "
            f"please use ',' to separate each float value."
        ) from e
```

File: deepmd/main.py [hunk 4]
```diff
@@ -659,6 +671,70 @@ def main_parser() -> argparse.ArgumentParser:
...
+        "--system",
+        default=".",
+        type=str,
+        help="The system dir. Recursively detect systems in this directory",
+    )
+    parser_change_bias_source.add_argument(
+        "-b",
+        "--bias-value",
+        default=None,
+        type=float_list,
```
Comment (njzjz) [inline deepmd/main.py:702]: I am wondering why not using `type=float, nargs='+'`. An example should be given, anyway.

File: deepmd/main.py (hunk 3 above)
Comment (njzjz) [reply deepmd/main.py:63]: We should turn on [`B904`](https://docs.astral.sh/ruff/rules/raise-without-from-inside-except/)

**Review by wanghan-iapcm — APPROVED** [review]

File: doc/model/change-bias.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+:::
+
+:::{tab-item} Changing bias using user input for **energy model**:
+
+```sh
+dp --pt change-bias model.pt -b -92.523,-187.66 -o model_updated.pt
+```
+
+Here, `-b` specifies user-defined energy bias for each type, separated by `,`,
```
Comment (njzjz) [inline doc/model/change-bias.md:37]: separated by space?

File: doc/model/index.rst
Comment (njzjz) [inline doc/model/index.rst:1]: Should we put in the model section or the training section?

Comment (iProzd) [reply doc/model/index.rst:1]: It's more about changing an attribute of model without training. Maybe model section is better?

Comment (njzjz) [reply doc/model/index.rst:1]: At least, I don't understand why it is ahead of `train-energy`.

Comment (iProzd) [reply doc/model/index.rst:1]: Oh I got it, I've put `change-bias` the same place with `sel`. It seems that `sel` indeed follows the descriptor sections while `change-bias` should follow all the model sections. Updated.

**Review by njzjz — APPROVED** [review]

**Pull request merged**